web/src/components/OutputTemplate/NewOutputTemplate/NewOutputTemplate.tsx

```diff
@@ -0,0 +1,138 @@
+import type {
+  CreateOutputTemplateMutation,
+  CreateOutputTemplateInput,
+  CreateOutputTemplateMutationVariables,
+} from 'types/graphql'
+
+import { navigate, routes } from '@redwoodjs/router'
+import { useMutation } from '@redwoodjs/web'
+import type { TypedDocumentNode } from '@redwoodjs/web'
+import { toast } from '@redwoodjs/web/toast'
+
+import OutputTemplateForm from 'src/components/OutputTemplate/OutputTemplateForm'
+
+const CREATE_OUTPUT_TEMPLATE_MUTATION: TypedDocumentNode<
+  CreateOutputTemplateMutation,
+  CreateOutputTemplateMutationVariables
+> = gql`
+  mutation CreateOutputTemplateMutation($input: CreateOutputTemplateInput!) {
+    createOutputTemplate(input: $input) {
+      outputTemplate {
+        id
+      }
+      signedUrls {
+        CPF1A
+        CPF1B
+        CPF1C
+        CPFSubrecipient
+      }
+    }
+  }
+`
+
+type ExtraFileFields = {
+  fileCPF1A: File[]
+  fileCPF1B: File[]
+  fileCPF1C: File[]
+  fileCPFSubrecipient: File[]
+}
+
+type FileNames = {
+  filenames: {
+    CPF1A: string
+    CPF1B: string
+    CPF1C: string
+    CPFSubrecipient: string
+  }
+}
+
+const NewOutputTemplate = () => {
+  const [createOutputTemplate, { loading, error }] = useMutation(
+    CREATE_OUTPUT_TEMPLATE_MUTATION,
+    {
+      onCompleted: () => {
+        toast.success('OutputTemplate created')
+        navigate(routes.outputTemplates())
+      },
+      onError: (error) => {
+        toast.error(error.message)
+      },
+    }
+  )
+
+  const putObject = async (signedUrl: string, file: File) => {
+    fetch(signedUrl, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': file.type,
+        'x-amz-server-side-encryption': 'AES256',
+      },
+      body: file,
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`)
+        }
+      })
+      .then((responseData) => {
+        console.log('File upload successful. Response:', responseData)
+      })
+      .catch((error) => {
+        console.error('Error uploading file:', error)
+      })
+  }
+  const verifyFileNames = (
+    input: CreateOutputTemplateInput & ExtraFileFields & FileNames
+  ) => {
+    const { filenames } = input
+    return (
+      filenames.CPF1A.includes('CPF1ABroadbandInfrastructureTemplate') &&
```

Comment:
Nit: I know these are supposed to be non null but might be nice to do the optional way here like `filenames.CPF1A?.includes` just to be safe so it throws the right error if they are undefined?